.circleci/config.yml

```diff
@@ -19,6 +17,7 @@ jobs:
       - run: greenkeeper-lockfile-update
       - run: npm ci
       - run: greenkeeper-lockfile-upload
+      - run: bash ./compile.sh
```

Comment:
This does not work as expected, because running `bash` creates a new subshell with its own environment variables. The use of `export xxx=...` will change the xxx environment variable as long as that script runs, but won't affect the environment variables outside of it.

(in general)
To get the desired behavior, the shell script should be sourced. Usually something like `source compile.sh` instead of `bash compile.sh`. Then the commands from the file will be read and executed in the current subshell.

(in the special case of CircleCI)
According to CircleCI's documentation, it sources `$BASH_ENV` before each step (Ctrl-F `$BASH_ENV` in https://circleci.com/docs/2.0/env-vars/ ). You can take advantage of this by prepending a new step (right after `steps`) that generates the environment variable, as follows:

```yaml
- run: bash generate-env.sh >> $BASH_ENV
```

`generate-env.sh` could be based on what you have right now, except `exit 0` being replaced with the following, and `export` removed from your original code, because it is used as a shell variable instead of an environment variable:

```sh
...
elif [ "${CIRCLE_BRANCH}" == "production" ]
then
  SITE_URL="https://color.firefox.com/"
  ADDON_URL="https://addons.mozilla.org/firefox/addon/firefox-color/"
else
  echo "Unknown branch: ${CIRCLE_BRANCH}" >&2
  exit 1
fi

# This will be appended to $BASH_ENV in .circleci/config.yml
cat <<ENV_HERE
export SITE_URL="${SITE_URL}"
export ADDON_URL="${ADDON_URL}"
ENV_HERE
```